skglm/experimental/quantile_huber.py

```diff
@@ -0,0 +1,206 @@
+        for i in range(n_samples):
+            residual = y[i] - Xw[i]
+            update -= self._grad_per_sample(residual)
+        return update / n_samples
+
+
+class SmoothQuantileRegressor(BaseEstimator, RegressorMixin):
+    """Quantile regression with progressive smoothing."""
+
+    def __init__(self, quantile=0.5, alpha=0.1, delta_init=1.0, delta_final=1e-3,
+                 n_deltas=10, max_iter=1000, tol=1e-4, verbose=False,
+                 solver="AndersonCD", fit_intercept=True):
+        self.quantile = quantile
+        self.alpha = alpha
+        self.delta_init = delta_init
+        self.delta_final = delta_final
+        self.n_deltas = n_deltas
+        self.max_iter = max_iter
+        self.tol = tol
+        self.verbose = verbose
+        self.solver = solver
+        self.fit_intercept = fit_intercept
+
+    def fit(self, X, y):
+        """Fit using progressive smoothing: delta_init --> delta_final."""
+        w = np.zeros(X.shape[1])
+        deltas = np.geomspace(self.delta_init, self.delta_final, self.n_deltas)
+
+        if self.verbose:
+            print(
+                f"Progressive smoothing: delta {self.delta_init:.3f} --> "
+                f"{self.delta_final:.3f} in {self.n_deltas} steps")
+
+        datafit = QuantileHuber(quantile=self.quantile, delta=self.delta_init)
+        penalty = L1(alpha=self.alpha)
+        # Solver selection
+        if isinstance(self.solver, str):
+            if self.solver == "FISTA":
+                if self.fit_intercept:
+                    import warnings
+                    warnings.warn(
+                        "FISTA solver does not support intercept. "
+                        "Falling back to fit_intercept=False."
+                    )
+                    self.fit_intercept = False
+                solver = FISTA(max_iter=self.max_iter, tol=self.tol)
+                solver.warm_start = True
+            elif self.solver == "AndersonCD":
+                solver = AndersonCD(max_iter=self.max_iter, tol=self.tol,
+                                    warm_start=True, fit_intercept=self.fit_intercept)
+            else:
+                raise ValueError(f"Unknown solver: {self.solver}")
+        else:
+            solver = self.solver
+
+        est = GeneralizedLinearEstimator(
+            datafit=datafit, penalty=penalty, solver=solver)
+
+        for i, delta in enumerate(deltas):
+            datafit.delta = float(delta)
+
+            est.fit(X, y)
+            w = est.coef_.copy()
+
+            if self.verbose:
+                residuals = y - X @ w
+                if self.fit_intercept:
+                    residuals -= est.intercept_
+                coverage = np.mean(residuals <= 0)
+                pinball_loss = np.mean(residuals * (self.quantile - (residuals < 0)))
+
+                print(
+                    f"  Stage {i+1:2d}: delta={delta:.4f}, "
+                    f"coverage={coverage:.3f}, pinball_loss={pinball_loss:.6f}, "
```

Comment:
no need to print coverage here, it's a statistical value, we're more interested in the value of delta and of the loss

Reply:
I'm -1 on this, we focus more on the convergence part ; if the model is not adapted to the data, it's not skglm's fault
What we want to check is only that we solve the optimization problem correctly (same for LienarRegression, we do not check if the residuals are homoscedastic)